conclave-common/src/main/kotlin/com/r3/conclave/common/internal/attestation/QuoteVerifier.kt

```diff
@@ -461,41 +461,14 @@ object QuoteVerifier {
     private fun isCpuSvnHigherOrEqual(pckTcb: IntArray, jsonTcb: Tcb): Boolean {
         for (j in pckTcb.indices) {
             // If *ANY* CPUSVN component is lower then CPUSVN is considered lower
-            if (pckTcb[j] < getSgxTcbComponentSvn(jsonTcb, j)) return false
+            if (pckTcb[j] < jsonTcb.sgxtcbcompsvn[j]) return false
         }
         // but for CPUSVN to be considered higher it requires that *EVERY* CPUSVN component to be higher or equal
         return true
     }
 
     private fun checkTcbLevel(pckExtensions: SGXExtensionASN1Parser, tcbInfo: TcbInfo): TcbStatus {
-        val tcbLevelStatus = getMatchingTcbLevel(pckExtensions, tcbInfo)
-        check(tcbInfo.version == 2 || tcbLevelStatus != TcbStatus.OutOfDateConfigurationNeeded) {
```

Comment:
The `tcbLevelStatus != TcbStatus.OutOfDateConfigurationNeeded` check is still valid, no?

Reply:
Nope, it's short-circuited by the || operation.

This did seem strange to me, was the original logic actually correct?

Reply:
What if the version is 3?

I'm pretty sure that's what the original code did, at least when I audited this code last. I can ask you check the upstream C code again with regards to this bit.

Reply:
I would like an answer to this.

Reply:
Wasn't able to find the original code. Do you know where I can find it?

Reply:
Never mind, was able to find it. Looks like there have been substantial changes on Intels side, might be a good idea to discuss offline.

Reply:
Sure. If there have been substantial changes upstream then we should probably review that as part of a separate ticket.

However, regardless of that, the check seems to be still valid is version can be either 2 or 3.

Reply:
Can you create a Jira ticket for the auditing the upstream changes.

Reply:
I've reintroduced the check and added some code to check for TDX. I've tried to mirror what's going on [here](https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteVerification/QVL/Src/AttestationLibrary/src/Verifiers/QuoteVerifier.cpp).